src/main/java/com/github/couchmove/repository/CustomSearchIndex.java

```diff
@@ -0,0 +1,45 @@
+package com.github.couchmove.repository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.couchbase.client.java.manager.search.SearchIndex;
+import com.couchbase.client.core.deps.com.fasterxml.jackson.annotation.JsonCreator;
+import com.couchbase.client.core.deps.com.fasterxml.jackson.annotation.JsonProperty;
+import com.couchbase.client.core.json.Mapper;
+
+public class CustomSearchIndex extends SearchIndex {
```

Comment:
Is it possible to use [Jackson MixIn annotations](https://github.com/FasterXML/jackson-docs/wiki/JacksonMixInAnnotations) instead ?

Reply:
The `toJson()`  function I overrode in this class is similar to the way the original Couchbase SearchIndex class is written. That is the function used in the Couchbase code to execute the REST request to update the FTS index, and it returns a String. I'm not sure if/how it would be possible to replicate this functionality with MixIn annotations instead, so I went with this.